### PR TITLE
move common raise_backend_exception method to base service class

### PR DESCRIPTION
**Pull request by LindseySaari**
## Description of change
This change extracts duplicate methods to a single method definition in the Common::Client::Base class for services. To follow best practices, the common code was moved to the base service class.

## Testing done
Test suite
Manual review of all service classes referencing raise_backend_exception
Review of service class inheritance from base service class

## Acceptance Criteria (Definition of Done)
- [x] Re-run of test suite
- [x] Investigation of current tests to verify that the exception raising case is still covered (Common::Exceptions::BackendServiceException)

#### Applies to all PRs

- [ ] Appropriate logging
- [ ] Swagger docs have been updated, if applicable
- [ ] Provide link to originating GitHub issue, or connected to it via ZenHub
- [ ] Does not contain any sensitive information (i.e. PII/credentials/internal URLs/etc., in logging, hardcoded, or in specs)
- [ ] Provide which alerts would indicate a problem with this functionality (if applicable)

### CODE REVIEW

File: lib/evss/service.rb [hunk 1]
```diff
@@ -65,25 +65,16 @@ def handle_error(error)
 
       case error
       when Faraday::ParsingError
-        raise_backend_exception('EVSS502', self.class)
+        raise_backend_exception('EVSS502', "EVSS::#{self.class}")
```
Comment (jholton) [inline lib/evss/service.rb]: The `EVSS::` prefix isn't necessary.  This results in `EVSS::EVSS::Service` being sent.

File: lib/common/client/base.rb [hunk 2]
```diff
@@ -54,6 +54,15 @@ def self.configuration(configuration = nil)
         @configuration ||= configuration.instance
       end
 
+      def raise_backend_exception(key, source, error = nil)
+        raise Common::Exceptions::BackendServiceException.new(
+          key,
+          { source: source.to_s },
+          error&.status,
```
Comment (johnpaulashenfelter) [inline lib/common/client/base.rb]: Just curious, why would error not have a status or body?

File: lib/common/client/base.rb [hunk 3]
```diff
@@ -54,6 +54,15 @@ def self.configuration(configuration = nil)
         @configuration ||= configuration.instance
       end
 
+      def raise_backend_exception(key, source, error = nil)
```
Comment (johnpaulashenfelter) [inline lib/common/client/base.rb]: I don't think there's a real plan around it, but how do we feel about keyword arguments ? Nothing you have to do here @LindseySaari just saw the method and was like -- oh, keywords.

File: lib/common/client/base.rb (hunk 2 above)
Comment (johnpaulashenfelter) [reply lib/common/client/base.rb]: Ah -- probably just from extracting the method.

File: lib/common/client/base.rb (hunk 3 above)
Comment (LindseySaari) [reply lib/common/client/base.rb]: Are you referring to using an options hash instead or basically running into issues where arguments with the same name as keywords can be dangerous in terms of being overridden?

Comment (johnpaulashenfelter) [reply lib/common/client/base.rb]: @LindseySaari I mean literal keyword arguments

`def raise_backend_exception(key:, source:, error: nil)`

this has nothing to do directly with your PR! This is just reviewing one method and focusing on one method made me wonder about moving towards keywords. Nothing actionable to do

Comment (LindseySaari) [reply lib/common/client/base.rb]: Right, I completely agree!

File: lib/evss/service.rb (hunk 1 above)
Comment (LindseySaari) [reply lib/evss/service.rb]: Fixing this 😄

**Pull request merged**